jsettlers.mapcreator/src/jsettlers/mapcreator/main/DataTester.java

```diff
@@ -49,21 +52,48 @@
 	private final ErrorList errorList;
 	private ArrayList<Error> errors = new ArrayList<Error>();
 
+	/**
+	 * Thread for testing
+	 */
+	private Thread thread;
+
+	/**
+	 * Running flag
+	 */
+	private boolean running = true;
+
 	public DataTester(MapData data, TestResultReceiver receiver) {
 		this.data = data;
 		this.receiver = receiver;
 		errorList = new ErrorList();
 	}
 
-	public void start() {
-		new Thread(this, "data tester").start();
+	public synchronized void start() {
+		running = true;
+		this.thread = new Thread(this, "data tester");
+		thread.start();
+	}
+
+	/**
+	 * Release all resources
+	 */
+	public void dispose() {
+		running = false;
+		synchronized (this) {
+			notifyAll();
+		}
+		try {
+			thread.join(100);
```

Comment:
System.exit is also very bad...
This is not my code. I only add the dipose().

For the moment it's working.
Can we do later a refactoring, and use an ExecutorService service instead?
Then we don't have the problem. 
Then also the whole synchronisation stuff can be removed, if there a new check should be done, we create a new job, which is executed in the Threadpool (single threaded).

OK for you?

Off topic, but I hope you read this here:
@michaelzangl you already have 144 commits on this project, you may know it:
Was there already a discussion about error loggin and exception handling?

I added a default thread exception handler in the main of the Map Editor, which catches all unhandled exception, display a popup and print the stacktrace to stderr.

I also used this popup on all important catch blocks I found, before only the exception was logged, you clocked somewhere and nothing happend. Really annoying as user.

I would generally show an error if an exception happens which matters.

And I would prefer to use SLF4J, you have a lot more possibilities (turn off for performance, detailed logging to stdout on debugging etc.)

I didn't see any discussion about this topic...